### PR TITLE
[stable/datadog] Fix several aspect of KSM integration

### LINKED ISSUE #15526: [stable/datadog] error in readme configuration parameter
**Describe the bug**
Configuration parameter in `rbac.serviceAccount` in readme should be `rbac.serviceAccountName` according to the code at https://github.com/helm/charts/blob/9b89f45a27aa98551ed01d60b362060f3271b7ff/stable/datadog/templates/daemonset.yaml#L128

**Which chart**:
datadog

### PR DESCRIPTION
**Pull request by clamoriniere**
#### What this PR does / why we need it:

this PR include several improvements around the `stable/kube-state-metrics` chart integration as `stable/datadog` chat's dependency. 
Also fix another typo in `Readme.md`

- cherry-pick: #15527 that fix  #15526
- update `kube-state-metrics` chart version to `1.2.0` corresponding to the `v1.5.0`
- fix `kube-state-metrics` values overwrite in the `datadog` values.yaml` and `Readme.md`  

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/datadog/values.yaml [hunk 1]
```diff
@@ -401,10 +401,16 @@ kube-state-metrics:
     #
     create: true
 
-    ## @param serviceAccountName - string - required
-    ## Ignored if rbac.create is true
-    #
-    serviceAccountName: default
```
Comment (CharlyF) [inline stable/datadog/values.yaml:407]: Why removing this ? we are using `.Values.rbac.serviceAccountName` not `.Values.rbac.serviceAccount.name`
in the different manifests.
For instance: https://github.com/helm/charts/blob/master/stable/datadog/templates/cluster-agent-deployment.yaml#L164

**Review by CharlyF — CHANGES_REQUESTED** [review]
I think the default value is not correct

File: stable/datadog/values.yaml (hunk 1 above)
Comment (clamoriniere) [reply stable/datadog/values.yaml:407]: the full path of this variable it is: `kube-state-metrics.rbac.serviceAccountName`  and not `.Values.rbac.create` as we use in your link.

In `kube-state-metrics` chart, they replace this variable by `.Value.rbac.serviceAccount.name`. That is why I added the following fields

**Pull request merged**